src/constraint_system/composer.rs

```diff
@@ -417,6 +417,112 @@ impl StandardComposer {
         self.n += 1;
     }
 
+    /// Utility function that allows to check on the "front-end"
+    /// side of the PLONK implementation if the identity polynomial
+    /// is satisfied for each one of the [`StandardComposer`]'s gates.
+    ///
+    /// The recommended usage is to derive the std output and the std error to a
+    /// text file and analyze there the gates.
+    ///
+    /// # Panic
+    /// The function by itself will print each circuit gate info until one of
+    /// the gates does not satisfy the equation or there are no more gates. If
+    /// the cause is an unsatisfied gate equation, the function will panic.
+    pub fn check_circuit_satisfied(&self) {
```

Comment:
This was already in PLONK, so I'm not sure how it happened that this suddenly got added as new. I guess because the branches diverged before this feature was included? If possible, it may be better to leave this out, since you will get a conflict out of it when trying to rebase on master, and you'll probably inadvertently remove the original author from these lines.

Reply:
It's better to 'allow incoming' when this goes into master. 
They were moved when the features wouldn't work with the new proof system, by another developer.
I'll make a note of allowing master to overwrite these lines.